primitives/transaction/src/lib.rs

```diff
@@ -432,6 +437,133 @@ impl Transaction {
     pub fn recipient(&self) -> &Address {
         &self.recipient
     }
+
+    pub fn related_addresses(&self) -> HashSet<Address> {
+        let mut set = HashSet::new();
+        set.insert(self.sender.clone());
+        set.insert(self.recipient.clone());
+
+        if let Ok(proof) = SignatureProof::deserialize_from_vec(&self.proof) {
+            set.insert(proof.compute_signer());
+        }
+
+        if self.recipient_type == AccountType::Vesting {
```

Comment:
An exhaustive `match` over the account types might be better because it makes sure we cover all cases (if we add some in the future).